packages/server/billing/helpers/processInvoiceItemHook.ts

```diff
@@ -69,6 +69,16 @@ const processInvoiceItemHook = async (stripeSubscriptionId: string) => {
     return
   }
   const {id: hookId, type, prorationDate, isProrated, orgId, userId} = hook
+
+  const orgUsers = await r
```

Comment:
I haven't been able to repro the mismatch error yet after following the testing instructions. Just wondering if you're able to consistently reproduce it?

I think the bug could well be the result of a race condition where multiple users accept the org invite at the same time, but we do have a Redis Lock on line 45. Wouldn't that prevent this function from being called at the exact same time by two different users?

If we do want to move the `orgUsers` query, would we want to move it to the same `Promise.all` where we're getting the `stripeSubscriptionItem`?

Reply:
> I haven't been able to repro the mismatch error yet after following the testing instructions. Just wondering if you're able to consistently reproduce it?

Yeah, I always able to reproduce it. I recorded a loom: https://www.loom.com/share/a4b0a054743f441781c31c9a5e950074

Did you follow my testing instruction in `master` branch?

> Wouldn't that prevent this function from being called at the exact same time by two different users?

@nickoferrall we only set lock for `3000` ms . The issue occurs if the function executes longer that time.

> If we do want to move the orgUsers query, would we want to move it to the same Promise.all where we're getting the stripeSubscriptionItem?

Make sense, moved to Promise.all